sdk/resource/config.go

```diff
@@ -71,6 +71,11 @@ func WithHost() Option {
 	return WithDetectors(host{})
 }
 
+// WithHostID adds host ID information to the configured resource.
+func WithHostID() Option {
+	return WithDetectors(hostIDDetector{})
```

Comment:
Shouldn't we add `hostIDDetector` to `WithHost()`? We might consider adding `WithHostName` to offer a resource detector that only sets `host.name` resource attribute.

Reply:
Because the rules for detecting HostID are different for containerized vs non-containerized hosts I didn't want to couple the `hostIDDetector` to `WithHost`, although I had considered it. My thoughts are that someone who wanted to use `WithHost` on a cloud provider wouldn't have to worry about detector order when detecting `HostID`. However, I could see having `WithHost` that detects `HostID` and `HostName`, and separate options `WithHostID` and `WithHostName` to collect the attributes individually. Let me know what you preference is.